src/core_plugins/region_map/public/region_map_visualization.js

```diff
@@ -156,13 +153,15 @@ export function RegionMapsVisualizationProvider(Private, Notifier, config) {
         const rowIndex = this._chartData.tables[0].rows.findIndex(row => row[0] === event);
         this._vis.API.events.addFilter(this._chartData.tables[0], 0, rowIndex, event);
       });
+
       this._choroplethLayer.on('styleChanged', (event) => {
         const shouldShowWarning = this._vis.params.isDisplayWarning && config.get('visualization:regionmap:showWarnings');
         if (event.mismatches.length > 0 && shouldShowWarning) {
-          this._notify.warning(`Could not show ${event.mismatches.length} ${event.mismatches.length > 1 ? 'results' : 'result'} on the map.`
-            + ` To avoid this, ensure that each term can be matched to a corresponding shape on that shape's join field.`
-            + ` Could not match following terms: ${event.mismatches.join(',')}`
-          );
+          toastNotifications.addWarning({
+            title: `Couldn't show ${event.mismatches.length} ${event.mismatches.length > 1 ? 'results' : 'result'} on the map`,
+            text: `You can fix this by ensuring that each of these terms can be matched to a corresponding shape `
+              + `on that shape's join field:${event.mismatches.join(',')}`
+          });
```

Comment:
Unable to show 3 results on map

Ensure that each term matches a shape on that shape's join field.